### PR TITLE
[Docs] Japanese translation of docs/proton_c_conversion.md

### PR DESCRIPTION
**Pull request by mtei**
## Description

Add a Japanese translation of docs/proton_c_conversion.md

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [x] Documentation

## Checklist

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by mtei** [comment]
@umi-umi @shelaf @s-show @hsgw @satt99 @takashiski @yoshihara よかったらレビューお願いします。

File: docs/ja/proton_c_conversion.md [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+#ifdef CONVERT_TO_PROTON_C
+    // Proton C code
+#else
+    // Pro Micro code
+#endif
+```
+
+`PORTB/DDRB` などが定義されていないというエラーが発生した場合は、ARM と AVR の両方で機能する [GPIO Controls](ja/internals_gpio_control.md) を使用するようにキーボードのコードを変換する必要があります。これは AVR ビルドにまったく影響を与えません。
+
+Proton C には1つのオンボードLED(C13)しかなく、デフォルトでは TXLED(D5) がそれにマップされています。代わりに RXLED(B0) をそれにマッピングしたい場合は、`config.h` に次のように追加してください。
```
Comment (shelaf) [inline docs/ja/proton_c_conversion.md:32]:
```suggestion
Proton C には1つのオンボード LED(C13)しかなく、デフォルトでは TXLED(D5) がそれにマップされています。代わりに RXLED(B0) をそれにマッピングしたい場合は、`config.h` に次のように追加してください。
```

File: docs/ja/proton_c_conversion.md [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+## 機能の変換
+
+下記は ARM ボードに実装されているものに基づいたデフォルトです。
+
+| 機能                                  | 説明                                                                               |
+|--------------------------------------|------------------------------------------------------------------------------------|
+| [オーディオ](ja/feature_audio.md)      | 有効                                                                      |
+| [RGB ライト](ja/feature_rgblight.md)  | 無効                                                                       |
+| [バックライト](feature_backlight.md)   | ARMが自動コンフィギュレーションを提供できるようになるまで、[タスク駆動 PWM](ja/(feature_backlight.md#software-pwm-driver))が強制されます |
+| USB Host (e.g. USB-USB converter)   | 未サポート (USB ホストコードは AVR 固有のもので、現在 ARM ではサポートされていません。 |
```
Comment (shelaf) [inline docs/ja/proton_c_conversion.md:45]:
```suggestion
| USB ホスト (例えば USB-USB コンバータ)   | 未サポート (USB ホストコードは AVR 固有のもので、現在 ARM ではサポートされていません。 |
```

File: docs/ja/proton_c_conversion.md [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+
+## 機能の変換
+
+下記は ARM ボードに実装されているものに基づいたデフォルトです。
+
+| 機能                                  | 説明                                                                               |
+|--------------------------------------|------------------------------------------------------------------------------------|
+| [オーディオ](ja/feature_audio.md)      | 有効                                                                      |
+| [RGB ライト](ja/feature_rgblight.md)  | 無効                                                                       |
+| [バックライト](feature_backlight.md)   | ARMが自動コンフィギュレーションを提供できるようになるまで、[タスク駆動 PWM](ja/(feature_backlight.md#software-pwm-driver))が強制されます |
```
Comment (shelaf) [inline docs/ja/proton_c_conversion.md:44]:
```suggestion
| [バックライト](feature_backlight.md)   | ARM が自動コンフィギュレーションを提供できるようになるまで、[タスク駆動 PWM](ja/(feature_backlight.md#software-pwm-driver))が強制されます |
```

File: docs/ja/proton_c_conversion.md [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+```
+MCU = STM32F303
+```
+
+次の変数が存在する場合は削除します。
+
+* `BOOTLOADER`
+* `EXTRA_FLAGS`
+
+Finally convert all pin assignments in `config.h` to the stm32 equivalents.
```
Comment (shelaf) [inline docs/ja/proton_c_conversion.md:61]:
```suggestion
```

File: docs/ja/proton_c_conversion.md [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+| 左側 | | 右側 |
+|------|-|-------|
+| `A4`<sup>3</sup> | | `B10` |
+| `A5`<sup>4</sup> | | `B11` |
+| `A6` | | `B12` |
+| `A7` | | `A14`<sup>5</sup> (SWCLK) |
+| `A8` | | `A13`<sup>5</sup> (SWDIO) |
+| `A15` | | RESET<sup>6</sup> |
+
+ノート:
```
Comment (shelaf) [inline docs/ja/proton_c_conversion.md:91]:
```suggestion
注釈:
```

File: docs/ja/proton_c_conversion.md [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+
+1. Pro Micro の Vcc は 3.3V または 5V にすることができます。
+2. Proton C のオンボード LED は、Pro Micro のように 2 つはありません、1つだけです。Pro Micro には、RX LED(`D5`) と TX LED(`B0`)があります。
```
Comment (shelaf) [inline docs/ja/proton_c_conversion.md:94]:
```suggestion
2. Proton C のオンボード LED は、Pro Micro のように2つはありません、1つだけです。Pro Micro には、RX LED(`D5`) と TX LED(`B0`)があります。
```

File: docs/ja/proton_c_conversion.md [hunk 7]
```diff
@@ -0,0 +1,99 @@
...
+
+下記は ARM ボードに実装されているものに基づいたデフォルトです。
+
+| 機能                                  | 説明                                                                               |
+|--------------------------------------|------------------------------------------------------------------------------------|
+| [オーディオ](ja/feature_audio.md)      | 有効                                                                      |
+| [RGB ライト](ja/feature_rgblight.md)  | 無効                                                                       |
+| [バックライト](feature_backlight.md)   | ARMが自動コンフィギュレーションを提供できるようになるまで、[タスク駆動 PWM](ja/(feature_backlight.md#software-pwm-driver))が強制されます |
+| USB Host (e.g. USB-USB converter)   | 未サポート (USB ホストコードは AVR 固有のもので、現在 ARM ではサポートされていません。 |
+| [分割キーボード](ja/feature_split_keyboard.md) | 部分的 - 有効にする機能に大きく依存する                                 |
```
Comment (shelaf) [inline docs/ja/proton_c_conversion.md:46]:
```suggestion
| [分割キーボード](ja/feature_split_keyboard.md) | 部分的 - 有効にする機能に大きく依存します                               |
```

File: docs/ja/proton_c_conversion.md [hunk 8]
```diff
@@ -0,0 +1,99 @@
...
+
+```c
+#ifdef CONVERT_TO_PROTON_C
+    // Proton C code
+#else
+    // Pro Micro code
+#endif
+```
+
+`PORTB/DDRB` などが定義されていないというエラーが発生した場合は、ARM と AVR の両方で機能する [GPIO Controls](ja/internals_gpio_control.md) を使用するようにキーボードのコードを変換する必要があります。これは AVR ビルドにまったく影響を与えません。
```
[the remark this answers is not in the document]
Comment (mtei) [reply docs/ja/proton_c_conversion.md:30]: 悩ましいです。
「GPIO 制御」「GPIO コントロール」
もっと意訳して、「GPIO 操作」

[the remark this answers is not in the document]
Comment (mtei) [reply docs/ja/proton_c_conversion.md:30]: #9751 で、タイトルに採用してたから、「制御」に統一しましょう、やっぱり。

**Comment by noroadsleft** [comment]
ありがとうございました！

**Pull request merged**